lib/ansible/modules/packaging/os/apt.py

```diff
@@ -854,9 +854,9 @@ def download(module, deb):
 
     try:
         rsp, info = fetch_url(module, deb, method='GET')
-        if info['status'] != 200:
-            module.fail_json(msg="Failed to download %s, %s" % (deb,
-                                                                info['msg']))
+        if info['status'] != 200 and not deb.startswith('file:/') and not (deb.startswith('ftp:/') and info.get('msg', '').startswith('OK')):
+            module.fail_json(msg="Download failed: {0}".format(deb), status_code=info['status'], response=info['msg'])
```

Comment:
Why add new single purpose `status_code` and `response` keys? I'd put that info in `msg`.

Reply:
I'd rather keep this module consistent than making it similar to `get_url`.